### DISCUSSION TITLE
GET /api/1/vehicles/{id}/vehicle_data and vehicle sleepiness

### DISCUSSION THREAD

**Discussion by SylvainGa** [timdorr/tesla-api#558] [Q&A]
Hi, I posted this within a thread and didn't get a response so I thought about making it its own post. I have a Garmin watch which polls the car's data every five minutes through this web request call:

https://owner-api.teslamotors.com/api/1/vehicles/{id}/vehicle_data

Now, does this call prevent it from sleeping? From what I read, it should, but below is seven sequences of such calls with 5 minutes between each call. For the first two, the car was awake and responded with 200 and valid data. At 15:30, it changed to a response code of 408 and no data was returned. It looks like the car fell asleep. Subsequent calls kept returning 408. Launching the Tesla app on my phone confirmed the car was asleep because it took over 10 seconds for the car to come online. Once online, it restarted responding with 200 and valid data. That seems to contradict what I've read. Am I missing something here? 
```
15:21:02 : onReceiveVehicleData responseCode is 200
15:26:01 : onReceiveVehicleData responseCode is 200
15:30:00 : onReceiveVehicleData responseCode is 408
15:36:00 : onReceiveVehicleData responseCode is 408
15:41:00 : onReceiveVehicleData responseCode is 408
15:46:02 : onReceiveVehicleData responseCode is 200
15:51:01 : onReceiveVehicleData responseCode is 200
```

Oh yeah, my Teslamate instance do not show 1-4 minutes charges that are usually shown when the car is plugged and it's prevented from sleeping (sentry on for example). That seems to confirm my assumption that the car falls asleep, even when polled with .../vehicle_data.

Thanks.

**Comment by core-hacked** [comment]
Just call the car to wake up using ``POST /api/1/vehicles/{id}/command/wake_up`` and wait a few seconds, then call the vehicle data endpoint.

> Late answer I know :/

**Comment by SylvainGa** [comment]
Thanks, but my question was the opposite :-) I don't want to wake the car and only get statuses when it's awake. Since I can only make a background process (ie, like MakeWebRequest) one per 5 minutes, I can't check if the car is asleep and query if not. Luckily, the "GET /api/1/vehicles/{id}/data_request/*_state" (and ../vehicle_data) commands do not wake the vehicle and simply returns error 408 when asleep. That's enough for me for my needs.

Thanks for taking the time to answer though :-)

**Reply by tdorssers** [reply]
../vehicle_summary gives you the vehicle state, to see if its asleep or not. You can also use api/1/vehicles/{vehicle_id}/latest_vehicle_data to get the cached data, from the time the vehicle was last seen.

**Reply by SylvainGa** [reply]
Wait, what? I knew this was coming but it's here already? I don't see it in the doc and a Google search revealed nothing. I gotta to give it a try! Thanks :-)

**Reply by SylvainGa** [reply]
Interesting, api/1/vehicles/{vehicle_id}/latest_vehicle_data did return 200 and valid data but, only when the car was awake. When asleep, I still got a 408 error :-(

**Reply by tdorssers** [reply]
For me it really returns cached data from the server, while the car is asleep. I am running 2022.12.3.2, maybe you run an older sw version?

**Reply by SylvainGa** [reply]
I'm at 2022.12.3.12. Odd that I'm seeing something different. These two debug calls shows what I'm seeing. First one was while the car was awake and second one was while it was asleep

```
20:34:44 : url is https://owner-api.teslamotors.com/api/1/vehicles/{myVehicleId}/latest_vehicle_data
20:34:44 : onReceiveVehicleData responseCode is 200

20:37:10 : url is https://owner-api.teslamotors.com/api/1/vehicles/{myVehicleId}/latest_vehicle_data
20:37:19 : onReceiveVehicleData responseCode is 408
```

**Reply by SylvainGa** [reply]
Did you have to do anything to make it work? Looking at the data returned when 408 is returned, I get

{error=>vehicle unavailable: {:error=>"vehicle unavailable:"}, error_description=>, response=>null}

I would love to get that call to work.

**Reply by tdorssers** [reply]
Nope it just works like /vehicle_data. Actually both /vehicle_data and /latest_vehicle_data behave the same for me: both return data while the car is sleeping. Don't know since when, but for years I got a 408 when the vehicle is asleep, but now I don't get a 408 anymore.

**Reply by SylvainGa** [reply]
Thanks, it gets weird. I'm doing the web requests through my Garmin Watch API and that's where I get the 408 error. Using Teslapi running on my RasberryPie, it works correctly and my car does return 200 with the data. Looking at the packets passed through both methods, it's the same but somehow through my watch, I get 408. I'll need to troubleshoot more, Thanks for your help.

**Reply by Panda88CO** [reply]
I think it is cached data - if you look at timestamp it does not change when you poll and it is sleeping
